sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-full-time-process-integration.scheduler.e2e-spec.ts

```diff
@@ -196,6 +196,24 @@ describe(
       // TODO Add other fields as needed.
     });
 
+    it("Should generate disbursement file with formatted records when there is no data for the ecert generation.", async () => {
+      // Queued job.
+      const { job } = mockBullJob<void>();
+
+      // Act
+      const result = await processor.processQueue(job);
+
+      // Assert uploaded file.
+      const uploadedFile = getUploadedFile(sftpClientMock);
+      const fileDate = dayjs().format("YYYYMMDD");
+      const uploadedFileName = `MSFT-Request\\DPBC.EDU.FTECERTS.${fileDate}.001`;
+      expect(uploadedFile.remoteFilePath).toBe(uploadedFileName);
+      expect(result).toStrictEqual([
+        `Generated file: ${uploadedFileName}`,
+        "Uploaded records: 0",
+      ]);
```

Comment:
Please also assert the formation of header and footer by adding the following code. 

```ts

const [header, footer] = uploadedFile.fileLines;
      // Validate header.
      expect(header).toContain("100BC  NEW ENTITLEMENT");
      // Validate footer.
      expect(footer.substring(0, 3)).toBe("999");

```